tests/functional/spec/features/submit_with_summary_return_to_answer.spec.js

```diff
@@ -0,0 +1,77 @@
+import HouseholdDetailsSummaryPage from "../../generated_pages/submit_with_summary_return_to_answer/house-details-section-summary.page.js";
+import SubmitPage from "../../generated_pages/submit_with_summary_return_to_answer/submit.page.js";
+import AddressDurationPage from "../../generated_pages/submit_with_summary_return_to_answer/address-duration.page.js";
+
+describe("Summary Anchor Scrolling", () => {
+  describe("Given I start a Test Section Summary survey", () => {
+    beforeEach(() => {
+      browser.openQuestionnaire("submit_with_summary_return_to_answer.json");
+      $(InsuranceTypePage.both()).click();
+      $(InsuranceTypePage.submit()).click();
+    });
+
+    it("When I have provided an answer and click through to the next question, Then the Previous link url shouldn't contain any anchors or reference to return_to or return_to_answer_id", () => {
+      expect($(InsuranceAddressPage.previous()).getAttribute("href")).not.to.contain("#");
+      expect($(InsuranceAddressPage.previous()).getAttribute("href")).not.to.contain("return_to");
+      expect($(InsuranceAddressPage.previous()).getAttribute("href")).not.to.contain("return_to_answer_id");
+    });
+
+    it("When I reach the section summary page, Then the Change link url should contain return_to, return_to_answer_id query params", () => {
+      $(InsuranceAddressPage.submit()).click();
+      $(AddressDurationPage.submit()).click();
+      expect($(PropertyDetailsSummaryPage.insuranceAddressAnswer2Edit()).getAttribute("href")).to.contain(
+        "?return_to=section-summary&return_to_answer_id=insurance-address-answer2#insurance-address-answer2"
+      );
+    });
+
+    it("When I edit an answer from the section summary page, Then the Previous link url should contain an anchor referencing the answer id of the answer I am changing", () => {
+      $(InsuranceAddressPage.submit()).click();
+      $(AddressDurationPage.submit()).click();
+      $(PropertyDetailsSummaryPage.insuranceAddressAnswer2Edit()).click();
+      expect($(InsuranceAddressPage.previous()).getAttribute("href")).to.contain("property-details-section/#insurance-address-answer2");
+    });
+
+    it("When I edit an answer from the section summary page and click the Previous link, Then the browser url should contain an anchor referencing the answer id of the answer I am changing", () => {
+      $(InsuranceAddressPage.submit()).click();
+      $(AddressDurationPage.submit()).click();
+      $(PropertyDetailsSummaryPage.insuranceAddressAnswer2Edit()).click();
+      $(InsuranceAddressPage.previous()).click();
+      expect(browser.getUrl()).to.contain("property-details-section/#insurance-address-answer2");
+    });
+
+    it("When I edit an answer from the section summary page and click the Submit button, Then the browser url should contain an anchor referencing the answer id of the answer I am changing", () => {
```

Comment:
Maybe for these to, the `Then` should be like `Then I am taken to the summary and the browser url contains ...`